pkg/devfile/parser/parse.go

```diff
@@ -212,11 +250,36 @@ func parseFromURI(uri string, curDevfileCtx devfileCtx.DevfileCtx) (DevfileObj,
 		d.Ctx = devfileCtx.NewURLDevfileCtx(u.String())
 	}
 	d.Ctx.SetURIMap(curDevfileCtx.GetURIMap())
-	// Fill the fields of DevfileCtx struct
-	err = d.Ctx.PopulateFromURL()
-	if err != nil {
-		return DevfileObj{}, err
+	return populateAndParseDevfile(d, true)
+}
+
+func parseFromRegistry(registryId, registryURL string, curDevfileCtx devfileCtx.DevfileCtx) (DevfileObj, error) {
+	if registryURL != "" {
+		devfileContent, err := getDevfileFromRegistry(registryId, registryURL)
+		if err != nil {
+			return DevfileObj{}, err
+		}
+		return ParseDevfile(ParserArgs{data: devfileContent, registryURLs: curDevfileCtx.GetRegistryURLs()})
+	} else if curDevfileCtx.GetRegistryURLs() != nil {
+		for _, registry := range curDevfileCtx.GetRegistryURLs() {
+			devfileContent, err := getDevfileFromRegistry(registryId, registry)
+			if devfileContent != nil && err == nil {
+				return ParseDevfile(ParserArgs{data: devfileContent, registryURLs: curDevfileCtx.GetRegistryURLs()})
+			}
+		}
+	} else {
+		return DevfileObj{}, fmt.Errorf("failed to parse from registry, registry URL is not provided")
 	}
-	return parseDevfile(d, true)
 
+	return DevfileObj{}, fmt.Errorf("failed to get registry id: %s from registry URLs provided", registryId)
```

Comment:
Think we can have one return at this point

`falied to parse parent.id %s, unable to find registry` something like that

Reply:
I think having two is more helpful when debugging? as the first one is because no registry URL is provided, and the second one is because cannot find parent id in URLs provided. 

But I'm fine with combining into one single error message if you think it's more clear